fluentdhec/lambda_function.py

```diff
@@ -17,113 +8,137 @@
 This takes kubernetes fluentd and HSM audit log events from cloudwatch
 logs and sends them to the splunk HEC
 """
+import os
+import gzip
+import json
+import base64
+import re
+from typing import Dict
+
+import dateparser
+
+import pyhec
+import hsmdecoder
 
 
-def lambda_handler(event, context):
-    if os.getenv('EVENT_DEBUG', '0') == '1':
+def lambda_handler(event: Dict, context) -> None:
+    """
+    Sends to the HEC every event that is not a healthcheck event.
+    """
+    if os.getenv("EVENT_DEBUG") == "1":
         print(event)
-    b64encoded_data = event['awslogs']['data']
+    b64encoded_data = event["awslogs"]["data"]
     compressed_data = base64.b64decode(b64encoded_data)
     uncompressed_data = gzip.decompress(compressed_data)
     data = json.loads(uncompressed_data)
-    for log_event in data['logEvents']:
+    for log_event in data["logEvents"]:
         if is_healthcheck(log_event):
             print("INFO: build_payload_k8s: dropping healthchecks")
             continue
         event = parse_log_event(log_event)
         event["source"] = context.function_name
-        event["host"] = data.get('logGroup', 'unknown')
+        event["host"] = data.get("logGroup", "unknown")
         if "time" not in event:
-            t = extract_time(log_event['message'])
-            if type(t) is int:
-                event["time"] = t
+            try:
+                event["time"] = extract_time(log_event["message"])
+            except Exception as e:
```

Comment:
Do we really want to catch `Exception`?

Reply:
We're mirroring the original functionality that hard-supressed errors like this.
Being more specific about exception handling is of course a good idea, but would require much more testing to ensure the same functionality, hence for time we skipped.

Reply:
This is a lightweight cleanup PR, we tried to resist logic changes.